compiler/src/iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenAttrs.td

```diff
@@ -434,4 +434,52 @@ def IREECodegen_EncodingNopLayoutAttr  :
   }];
 }
 
+//===---------------------------------------------------------------------===//
+// iree_codegen.rotate_rows
+//===---------------------------------------------------------------------===//
+
+def IREECodegen_RotateRowsAttr  :
+    AttrDef<IREECodegen_Dialect, "RotateRows", [
+    DeclareAttrInterfaceMethods<IREECodegen_SwizzleAttrInterface, [
+        "swizzleOffset",
+        "getAccessElementCount"
+      ]>
+    ]> {
+  let mnemonic = "rotate_rows";
+  let summary = "An attribute that describes a swizzling pattern for rotating rows.";
+  let description = [{
+    This attribute rotates accesses of |access_width| within rows of size
+    |row_width|. For any given access into logical memref of shape
+    `memref<...xNx|access_width|x!eltype>` where `N = row_width / access_width`
+    at position `(i, j, 0)` is rotated to `(i, (i + j) % N, 0)`. For example,
```

Comment:
This looks great. I assume the indexing maps can have more leading dims?

Reply:
Yes, leading dims are effectively all collapsed into `i` here. That's what the `...` is trying to indicate.